### PR TITLE
fix(autocomponents): fix up package dependencies + add vite int. test

### PR DESCRIPTION
**Pull request by danroberts**
We allowed a few packaging issues to slip through, which caused errors when importing from @gadgetinc/react/auto.

This fixes a few of those depednency issues and introduces an integration test which pacakges and installs @gadgetinc/react in a new vite project and imports AutoForm.

## PR Checklist

- [ ] Important or complicated code is tested
- [ ] Any user facing changes are documented in the Gadget-side changelog
- [ ] Any immediate changes are slated for release in Gadget via a generated package dependency bump
- [ ] Versions within this monorepo are matching and there's a valid upgrade path

### CODE REVIEW

**Comment by danroberts** [comment]
Superseded by https://github.com/gadget-inc/js-clients/pull/564

File: packages/react/package.json [hunk 1]
```diff
@@ -48,6 +51,8 @@
     "@gadgetinc/api-client-core": "^0.15.24",
     "@graphql-typed-document-node/core": "^3.2.0",
     "@hookform/resolvers": "^3.3.1",
+    "@types/tmp": "^0.2.6",
+    "copyfiles": "^2.4.1",
```
Comment (MillanWangGadget) [inline packages/react/package.json:55]: Out of curiousity, what goes into the decision of making these `dependencies` vs `devDependencies`. It that they are not referenced by prodution facing files

**Review by MillanWangGadget — APPROVED** [review]
- 🍏 No more shopify app bridge in PolarisAutoButton
- 🍏 The errors for missing CSS files from within this repo are gone upon gitpkg installation 
- 🍏 New vite tests look reasonable

**Comment by MillanWangGadget** [comment]
For the mdxEditor css file, would it be reasonable to make a copy in the repo and import it in the same way that these css flies are included? That way, the richText can truly become an optional dep

File: packages/react/package.json (hunk 1 above)
Comment (danroberts) [reply packages/react/package.json:55]: They should be devDependencies, good catch.

**Comment by danroberts** [comment]
> For the mdxEditor css file, would it be reasonable to make a copy in the repo and import it in the same way that these css flies are included? That way, the richText can truly become an optional dep

I'm not sure how this would work, but I think we should do a followup here to make the mdx editor more of a standalone optional dep -- right now, installing the @mdxeditor package as a peer dependency is still required.

**Review by airhorns — APPROVED** [review]
LGTM

**Review by alanko0511 — APPROVED** [review]

**Pull request merged**